keras/backend/mxnet_backend.py

```diff
@@ -3043,8 +3043,11 @@ def conv1d(x, kernel, strides=1, padding='valid',
 
     # Causal requires temporal padding.
     # MXNet backend does not support temporal padding on 3D tensor.
-    if padding is 'causal':
-        raise ValueError('MXNet Backend: conv1d does not support "causal" padding mode')
+    kernel_shape = kernel.shape
+    if padding == 'causal':
+        pad = dilation_rate * (kernel_shape[0] - 1)
+        x = temporal_padding(x, (pad, 0))
```

Comment:
MXNet temporal_padding supports only 3D and 4D tensor. Is it something we need to check here?

Reply:
we check for dimension size inside temporal_padding function - should we throw an error message here?

Reply:
I would check here can be handled more meaningfully. Because when user adds a Conv1D layer and gets temporal padding error, may not be directly telling what theissue is

Reply:
added check for dimensions inside conv1d